src/library_sockfs.js

```diff
@@ -280,6 +280,13 @@ mergeInto(LibraryManager.library, {
 
         function handleMessage(data) {
           assert(typeof data !== 'string' && data.byteLength !== undefined);  // must receive an ArrayBuffer
+
+          // An empty ArrayBuffer will emit a pseudo disconnect event
```

Comment:
I think that reasoning makes sense. It's hard for me to think of something wrong with it.

I guess it's not practical to add a test for this? If not, I'm to merge without.